Stop including shared library in java_wrap_cc runtime_deps.

SanjayVas opened the pull request:
This is a breaking change. It includes a Java util for loading a library from JAR resources.

The native library is included as a dep of the resulting java_library. According to the Bazel docs, this should result in a native shared library being included in the resulting JAR. According to bazelbuild/bazel#3079, this does not currently function as documented. Therefore, the macro generates a shared library target that the caller can include in JAR resources.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/world-federation-of-advertisers/rules_swig/4)
<!-- Reviewable:end -->

stevenwarejones reviewed (changes requested):
what tests do you have for this?

> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/rules_swig/4#-:-N3uqMnn1YnkypU1u8LJ:b-swjkxa)* status: 0 of 3 files reviewed, all discussions resolved



<!-- Sent from Reviewable.io -->

SanjayVas reviewed (commented):
None in this repo, unfortunately. I don't actually have any familiarity with actually writing SWIG. Happy to have a contribution of a simple SWIG example with a test that calls it using the loadLibrary util.

I manually tested this by modifying anysketch-java locally.

> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/rules_swig/4#-:-N3vaaUp3tdSIwnETKvJ:bf8e7o3)* status: 0 of 3 files reviewed, all discussions resolved



<!-- Sent from Reviewable.io -->

stevenwarejones reviewed (commented):
I could contribute a simple test. Would you want to merge this before that?

> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/rules_swig/4#-:-Nkb1LnL61OluSrW-LCS:bad2x79)* status: 0 of 3 files reviewed, all discussions resolved



<!-- Sent from Reviewable.io -->

SanjayVas reviewed (commented):
Obviously this isn't urgent given how long this has sat, so up to you on whether you'd like it to wait for that test. IIRC this was mainly intended to fix a design flaw that prevented these libraries from being included in Maven artifacts or used in remote execution environments such as Apache Beam.

> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/rules_swig/4#-:-Nkb6UoS6xx6wUUbhhFo:b-nyzla)* status: 0 of 3 files reviewed, all discussions resolved (waiting on @stevenwarejones)



<!-- Sent from Reviewable.io -->

stevenwarejones reviewed (commented):
yeah - I'm aware. I had thought the java portion would live under common-jvm - I just remembered there is this old PR too -- https://github.com/world-federation-of-advertisers/common-jvm/pull/120/files 

> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/rules_swig/4#-:-Nkb9g1_0X3hE7FJuKKT:bn1zqft)* status: 0 of 3 files reviewed, all discussions resolved



<!-- Sent from Reviewable.io -->

SanjayVas reviewed (commented):
> I had thought the java portion would live under common-jvm

The Java portion is tied to the implementation of the rule, so it makes more sense here.

> I just remembered there is this old PR too -- [https://github.com/world-federation-of-advertisers/common-jvm/pull/120/files](https://github.com/world-federation-of-advertisers/common-jvm/pull/120/files)

Yes. This PR was meant to avoid the need for a panelmatch-specific workaround.

> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/rules_swig/4#-:-NkbRsQQ6vFLH5ydWXzZ:b-jrrx60)* status: 0 of 3 files reviewed, all discussions resolved (waiting on @stevenwarejones)



<!-- Sent from Reviewable.io -->

stevenwarejones reviewed (approved):
> Reviewed 1 of 3 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/rules_swig/4)* status: 2 of 3 files reviewed, all discussions resolved



<!-- Sent from Reviewable.io -->

SanjayVas committed:
Stop including shared library in java_wrap_cc runtime_deps.

This is a breaking change. It includes a Java util for loading a library from JAR resources.

stevenwarejones reviewed (approved):
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/rules_swig/4)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @SanjayVas)



<!-- Sent from Reviewable.io -->

SanjayVas commented:
Dropping this in favor of allowing flexibility for where a library is loaded from. See world-federation-of-advertisers/common-jvm#256, world-federation-of-advertisers/common-jvm#257, and https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/main/src/main/kotlin/org/wfanet/measurement/eventdataprovider/shareshuffle/Native.kt
